app/hellodjango/locations/helpers/file_ops.py

```
@@ -0,0 +1,103 @@
+import pathlib
+import subprocess
+import zipfile
+
+from siege_utilities.files.hashing import calculate_file_hash
+
+logger = logging.getLogger("django")
+
+
+def run_subprocess(command_list):
+    """Run a shell command list; raises on non-zero exit code."""
+    p = subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)